### PR TITLE
Avoid temporary files created by the testsuite

### PR DESCRIPTION
**Pull request by wilzbach**
Output files should be stored in `test_results` ...

### CODE REVIEW

File: test/compilable/vcg-ast.d [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module vcg;
 // REQUIRED_ARGS: -vcg-ast -o-
 // PERMUTE_ARGS:
+// POST_SCRIPT: rm -f compilable/vcg-ast.d.cg
```
Comment (wilzbach) [inline test/compilable/vcg-ast.d]: This has already been approved at https://github.com/dlang/dmd/pull/6888

File: test/compilable/futurexf.d [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /* PERMUTE_ARGS:
-   REQUIRED_ARGS: -Xffuture.json
+   REQUIRED_ARGS: -Xf${RESULTS_DIR}/compilable/futurexf.out
```
Comment (JinShil) [inline test/compilable/futurexf.d]: Good!

**Review by JinShil — APPROVED** [review]

**Review by PetarKirov — APPROVED** [review]

File: test/compilable/vcg-ast.d [hunk 3]
```diff
@@ -1,6 +1,7 @@
 module vcg;
 // REQUIRED_ARGS: -vcg-ast -o-
 // PERMUTE_ARGS:
+// POST_SCRIPT: rm -f compilable/vcg-ast.d.cg && cat
```
Comment (wilzbach) [inline test/compilable/vcg-ast.d]: Looks like the temporary filename is always appended here :/

Comment (JinShil) [reply test/compilable/vcg-ast.d]: I though the `-f` was supposed to handle that.  Is `rm` aliased to something else on the auto-tester?

Comment (JinShil) [reply test/compilable/vcg-ast.d]: I think you'll have to write a `.sh` file for this and put it in `/extra-files/` to make it work on all platforms.

**Comment by wilzbach** [comment]
Closing in favor of #7482

**Pull request closed without merging**